hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/utils/TransactionUtils.java

```diff
@@ -67,7 +67,8 @@ public static Option<HoodieCommitMetadata> resolveWriteConflictIfAny(
       Option<HoodieInstant> lastCompletedTxnOwnerInstant,
       boolean reloadActiveTimeline,
       Set<String> pendingInstants) throws HoodieWriteConflictException {
-    if (config.getWriteConcurrencyMode().supportsOptimisticConcurrencyControl()) {
+    // Skip to resolve conflict if there are multiple writers working on the same MOR table with simple bucket index
+    if (config.getWriteConcurrencyMode().supportsOptimisticConcurrencyControl() && !config.isLocklessMultiWriter()) {
       // deal with pendingInstants
```

Comment:
In `HoodieFlinkWriteClient.preTxn`, we can also skip the metadata collection if the NB-CC is enabled.